docs: add notes on contents of Postgres DB as Key-Value store to Postgres.md

84adam opened the pull request:
## Change Description
Adding notes on how Postgres DB is currently used as a Key-Value store, and which tables are included.

Addresses Issue #6619 - "Missing documentation for postgresql database structure"

## Steps to Test
Key/Value pairs are not all TLV-encoded. Schema/structure is unclear (outside of K/V column names). However, some Key Names and Values are possible to decode into human-readable strings. For example: I used Python + psycopg2 + text processing to decode some of these, such as Node names (see comment on Gist): https://gist.github.com/84adam/d85f5f40e8821925585f263f68b655eb

SQL statements do not reveal any consistent encoding or structure:

```
select * from channeldb_kv limit 5;
                  key                   | value | parent_id | id | sequence 
----------------------------------------+-------+-----------+----+----------
 \x6f70656e2d6368616e2d6275636b6574     |       |           |  1 |         
 \x636c6f7365642d6368616e2d6275636b6574 |       |           |  2 |         
 \x636972637569742d6677642d6c6f67       |       |           |  3 |         
 \x6677642d7061636b61676573             |       |           |  4 |         
 \x696e766f69636573                     |       |           |  5 |         
```

```
select * from walletdb_kv limit 5;
        key         | value | parent_id | id | sequence 
--------------------+-------+-----------+----+----------
 \x77616464726d6772 |       |           |  1 |         
 \x7774786d6772     |       |           |  2 |         
 \x6d61696e         |       |         1 |  3 |         
 \x73796e63         |       |         1 |  4 |         
 \x73636f7065       |       |         1 |  5 | 
```

This documentation update attempts to make clear that users cannot expect consistency for now in issuing SQL queries to their Postgres databases, and points out where more documentation will be required once additional schema is introduced into these tables.

ellemouton commented on docs/postgres.md:
```diff
@@ -45,3 +45,38 @@ might be slow for unexpected reasons.
+## What is in the database?
+
+At present, the Postgres Database functions as a Key-Value Store, much as Bolt DB does.
+
```
nit: remove the new lines between sentences?

ellemouton reviewed (commented):
Thanks for the PR @84adam! 

Pls see our [contributor guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#model-git-commit-messages) for how to structure your commit title.

So would suggest something like: `docs: add <short description here>`

Also please add an entry to the release notes

84adam commented:
Thank you, @ellemouton. Updated.

yyforyongyu reviewed (changes requested):
Thanks for the PR! @ellemouton 's comment is not addressed thought,

> So would suggest something like: docs: add <short description here>

So the git commit messages need to be updated as suggested.

84adam commented:
Thanks, @yyforyongyu. I believe I have corrected this now.

ellemouton commented on docs/postgres.md:
```diff
@@ -48,13 +48,7 @@ In case a replication architecture is planned, streaming replication should be a
-More schema will be introduced over time.
-
-At present the schema for each table/relation is simply: `key`, `value`, `parent_id`, `id`, `sequence`.
+At present, the Postgres Database functions as a Key-Value Store, much as Bolt DB does. Some values are TLV-encoded while others are not. More schema will be introduced over time. At present the schema for each table/relation is simply: `key`, `value`, `parent_id`, `id`, `sequence`.
```
pls squash this into the previous commit :)

84adam commented:
@yyforyongyu I believe the requested changes have been completed. (Squashed commits.) Please let me know if I'm missing anything else. Thanks!

yyforyongyu commented on docs/postgres.md:
```diff
@@ -45,3 +45,32 @@ might be slow for unexpected reasons.
+
+Notably, Invoice DB is maintained separately alongside the LND node.
+
+References:
```
I don't think we need this section. My instinct is they are not "academic" enough to be used as references.

yyforyongyu commented on docs/postgres.md:
```diff
@@ -45,3 +45,32 @@ might be slow for unexpected reasons.
+- https://lightningcommunity.slack.com/archives/C03ECAP7HK6/p1675357998163889?thread_ts=1675357814.564459&cid=C03ECAP7HK6 - 'Values are a mix of TLV and non-TLV-encoded data. Key/Values are just stored as plain byte slices. With more schema introduced it'll gain a lot of performance over time. This methodology allows for incrementally upgrading the DB code over time. Invoice DB transactions do no ...
+- https://gist.github.com/84adam/d85f5f40e8821925585f263f68b655eb - Python script to decode keys/values from Postgres LND database
+
+TODO: Update this section once new schema are introduced.
```
nit: I think this `TODO` goes without saying and could be removed.

84adam commented:
Removed previous "references" and squashed commits.

```
- https://twitter.com/positiveblue2/status/1621170766808256512 - "Currently LND supports Postgres but it is used as K/V store. Introducing better schemas will unlock better performance and make easier to get insights from our nodes."
- https://twitter.com/lndreviewclub/status/1618696557402152961 - "Postgres support was added in lnd 0.14, but the next step is to convert the invoices to a native SQL schema."
- https://lightningcommunity.slack.com/archives/C03ECAP7HK6/p1675357998163889?thread_ts=1675357814.564459&cid=C03ECAP7HK6 - 'Values are a mix of TLV and non-TLV-encoded data. Key/Values are just stored as plain byte slices. With more schema introduced it'll gain a lot of performance over time. This methodology allows for incrementally upgrading the DB code over time. Invoice DB transactions do not touch data in other buckets (they are self contained).'
- https://gist.github.com/84adam/d85f5f40e8821925585f263f68b655eb - Python script to decode keys/values from Postgres LND database
```

yyforyongyu reviewed (approved):
LGTM✅ Thanks again for the PR!

ellemouton reviewed (approved):
Looks good! I think it could be worth adding an entry to sqlite.md too. Or at least linking to this explanation from there?

84adam commented:
Anything I'm missing for this PR? Curious if you have any idea when it might be merged? @ellemouton @yyforyongyu 

Thanks!

yyforyongyu commented:
@84adam added to `0.16.1` so this will be merged before that release.
